ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -200,24 +213,33 @@ public static SourceBuilder instance() {
     );
   }
 
-  private static PhysicalSchema getPhysicalSchemaWithKeyAndPseudoCols(
+  private static PhysicalSchema getPhysicalSchemaWithPseudoColumnsToMaterialize(
       final SourceStep<?> streamSource) {
 
     final boolean windowed = streamSource instanceof WindowedTableSource;
 
     final FormatInfo formatInfo = streamSource.getFormats().getKeyFormat();
     final SerdeFeatures serdeFeatures = streamSource.getFormats().getKeyFeatures();
+
     final KeyFormat keyFormat = windowed
         ? KeyFormat.windowed(
-            formatInfo, serdeFeatures, ((WindowedTableSource) streamSource).getWindowInfo())
+        formatInfo, serdeFeatures, ((WindowedTableSource) streamSource).getWindowInfo())
         : KeyFormat.nonWindowed(formatInfo, serdeFeatures);
 
-    final Formats formats = of(keyFormat, streamSource.getFormats().getValueFormat());
+    final Formats format = of(keyFormat, streamSource.getFormats().getValueFormat());
+
+    //we currently don't materialize partition and offset to get them downstream for windowed
```

Comment:
Am I misremembering or did you say that push queries _do_ allow windowed table joins, in which case we do need to materialize partition and offset for windowed tables, else the new pseudocolumns won't work with windowed joins?

Reply:
push queries do _not_, see https://github.com/confluentinc/ksql/issues/8055

Reply:
In that case why do we need to bump the execution step version for WindowedSourceTable at all? Can we just use the original WindowedSourceTableV1 and the original SourceBuilderV1 for the windowed table case to save on new code?

Reply:
We could. At the time I wrote https://github.com/confluentinc/ksql/pull/7990 I didn't know about the constraint on windowed tables. I first thought about exactly this when when I brought it up in our sync and you mentioned how rare windowed table joins are. Since we have no need to change the topology, we can definitely still use the V1 step. The tradeoff is that if we did want to support these joins in the future, someone would need to rewrite the V2 step. This is opposed to if we complete the windowed step as part of KLIP-50, which will give windowed tables a parallel topology change with unwindowed tables and allow for joins, or anything else where we need to access pseudocolumns in a state store.

This was a pretty big oversight by me to not bring it up in our sync, but I'd be interested to hear what you think about the best way to do this.

Reply:
I think it's preferable to not bump the execution step version, as doing so adds significant code. If joins between windowed tables haven't worked until now, I think the fact that we haven't heard any reports about it means no one is using them and it doesn't make sense for us to invest in supporting them in the near future. This is not surprising to me since a windowed join as a push query offers limited usefulness given that it cannot be converted into a persistent query at this time.

Reply:
As discussed offline, let's remove the unnecessary execution step (WindowedSourceTableV2) in a follow-up PR. Merging this one for now. Thanks @Sullivan-Patrick !